src/components/search/SearchList.tsx

```diff
@@ -0,0 +1,44 @@
+import { GET_PRODUCTS } from "@app/queries/products";
+import React from "react";
+import { View, Text, Image, ActivityIndicator } from "react-native";
+import { useQuery } from "@apollo/react-hooks";
+import { Product } from "../../../interfaces/products";
+interface SearchListProps {
+  searchText: string;
+}
+
+const SearchList = ({ searchText }: SearchListProps) => {
+  const { loading, error, data } = useQuery(GET_PRODUCTS, {
+    variables: { searchTerm: searchText, perPage: 4 },
+  });
+  let searchedItems;
+  if (data) {
+    searchedItems = data.ProductItems.items;
+  }
+  return (
+    <View className="absolute w-full -bottom-[228px] min-h-[228px]">
+      <View className="bg-slate-200 py-2.5 px-2">
+        {loading && <ActivityIndicator />}
+        {error && <Text>There was an error</Text>}
+        {searchedItems?.length == 0 && <Text>No result</Text>}
+        {data &&
+          searchedItems.map((item: Product) => (
```

Comment:
do you need to check `data`?
```suggestion
        {searchedItems?.map((item: Product) => (
```

Reply:
Yep. It throws an error if I don't check `data`
<img width="383" alt="image" src="https://user-images.githubusercontent.com/59636753/233751510-bada7821-7df8-498a-9df0-e2edf1bb18be.png">

Reply:
I said to use `?.`